kalliope/trigger/snowboy/snowboydecoder.py

```
@@ -88,14 +88,27 @@ def audio_callback(in_data, frame_count, time_info, status):
         self.ring_buffer = RingBuffer(
             self.detector.NumChannels() * self.detector.SampleRate() * 5)
         self.audio = pyaudio.PyAudio()
-        self.stream_in = self.audio.open(
-            input=True, output=False,
-            format=self.audio.get_format_from_width(
-                self.detector.BitsPerSample() / 8),
-            channels=self.detector.NumChannels(),
-            rate=self.detector.SampleRate(),
-            frames_per_buffer=2048,
-            stream_callback=audio_callback)
+        self.open_audio(audio_callback)
+
+    def open_audio(self, audio_callback, i=0):
+        try:
+            self.stream_in = self.audio.open(
+                input=True, output=False,
+                format=self.audio.get_format_from_width(
+                    self.detector.BitsPerSample() / 8),
+                channels=self.detector.NumChannels(),
+                rate=self.detector.SampleRate(),
+                frames_per_buffer=2048,
+                stream_callback=audio_callback)
+        except IOError:
+            logger.debug("IOError raised, i = %s" % i)
+            if i == 5:
+                # Let's give up...
+                raise
```

Can you please raise a new exception like "OpenAudioException" with a message ?